src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx

```diff
@@ -8,23 +8,26 @@ import React, { useState, useCallback, useMemo, useEffect } from 'react';
 import { EuiButton, EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
 import useObservable from 'react-use/lib/useObservable';
 import { CoreStart, WorkspaceAttribute } from '../../../../../core/public';
+import { useOpenSearchDashboards } from '../../../../../plugins/opensearch_dashboards_react/public';
 
 type WorkspaceOption = EuiComboBoxOptionOption<WorkspaceAttribute>;
 
 interface WorkspaceDropdownListProps {
   coreStart: CoreStart;
   onCreateWorkspace: () => void;
-  onSwitchWorkspace: (workspaceId: string) => Promise<void>;
 }
 
 function workspaceToOption(workspace: WorkspaceAttribute): WorkspaceOption {
   return { label: workspace.name, key: workspace.id, value: workspace };
 }
 
 export function WorkspaceDropdownList(props: WorkspaceDropdownListProps) {
-  const { coreStart, onCreateWorkspace, onSwitchWorkspace } = props;
+  const { coreStart, onCreateWorkspace } = props;
   const workspaceList = useObservable(coreStart.workspaces.client.workspaceList$, []);
   const currentWorkspaceId = useObservable(coreStart.workspaces.client.currentWorkspaceId$, '');
+  const {
+    services: { workspaces },
+  } = useOpenSearchDashboards();
```

Comment:
different place to get the same reference `workspaces`. The approach to get `workspaces` client should be consistent in a single component.